enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/EnterpriseAuthSubject.java

```diff
@@ -115,27 +105,27 @@ public boolean allowsSchemaWrites()
     @Override
     public String name()
     {
-        return subject.getPrincipal().toString();
+        return shiroSubject.getPrincipal().toString();
     }
 
-    Subject getSubject()
+    ShiroSubject getShiroSubject()
     {
-        return subject;
+        return shiroSubject;
     }
 
     private AccessMode.Static getAccessMode()
```

Comment:
Instead of returning an AccessMode, we could have a private isPermitted method that was called from the allowsX methods. Eg allowsWrites could just return isPermitted( READ_WRITE )